sunpy/net/helioviewer.py

```diff
@@ -127,29 +173,37 @@ def download_jp2(self, date, directory=None, overwrite=False, **kwargs):
         >>> import sunpy.map
         >>> from sunpy.net import helioviewer
         >>> hv = helioviewer.HelioviewerClient()  # doctest: +REMOTE_DATA
-        >>> filepath = hv.download_jp2('2012/07/03 14:30:00', observatory='SDO', instrument='AIA', detector='AIA', measurement='171')   # doctest: +REMOTE_DATA
+        >>> filepath = hv.download_jp2('2012/07/03 14:30:00', observatory='SDO', instrument='HMI', detector=None, measurement='continuum')   # doctest: +REMOTE_DATA
         >>> aia = sunpy.map.Map(filepath)   # doctest: +REMOTE_DATA
         >>> aia.peek()   # doctest: +SKIP
-
-        >>> data_sources = hv.get_data_sources()  # doctest: +REMOTE_DATA
-        >>> file = hv.download_jp2('2012/07/03 14:30:00', sourceId=data_sources['SOHO']['LASCO']['C2']['white-light']['sourceId'])   # doctest: +REMOTE_DATA
         """
+
+        if('observatory' in kwargs):
+            observatory = kwargs['observatory'] if(kwargs['observatory']) else None 
+            instrument = kwargs['instrument'] if(kwargs['instrument']) else None
+            detector = kwargs['detector'] if(kwargs['detector']) else None
+            measurement = kwargs['measurement'] if(kwargs['measurement']) else None
+            key = (observatory, instrument, detector, measurement)
+            sourceid = self.datasource_info[key]
```

Comment:
We can tidy this up now as well?

Reply:
So shall we make observatory instruments etc required arguments now? But I think we should not do that because even in the older version of API these parameters were not mandatory

Reply:
They should be. It was a mistake on the older version.

Reply:
Ok I'll do that